File: leetcode/1605-find-valid-matrix-given-row-and-column-sums/1605-find-valid-matrix-given-row-and-column-sums.cpp
class Solution {
public:
    vector<vector<int>> restoreMatrix(vector<int>& rowSum, vector<int>& colSum) {
        vector<vector<int>> res(rowSum.size(), vector<int>(colSum.size()));
        for (int i = 0; i < res.size(); i++) 
            for (int j = 0; j < res[0].size() && rowSum[i]; j++) 
                res[i][j] = min(rowSum[i], colSum[j]), 
                rowSum[i] -= res[i][j], 
                colSum[j] -= res[i][j];
        return res;
    }
};

File: 1605-find-valid-matrix-given-row-and-column-sums/1605-find-valid-matrix-given-row-and-column-sums.rs
impl Solution {
    pub fn restore_matrix(rs: Vec<i32>, cs: Vec<i32>) -> Vec<Vec<i32>> {
        let mut row_sum = rs;
        let mut col_sum = cs;
        let mut res: Vec<Vec<i32>> = vec![vec![0; col_sum.len()]; row_sum.len()];
        for i in 0..res.len() {
            for j in 0..res[0].len() {
                if row_sum[i] > 0 {
                    res[i][j] = std::cmp::min(row_sum[i], col_sum[j]);
                    row_sum[i] -= res[i][j];
                    col_sum[j] -= res[i][j];
                }
            }
        }
        res
    }
}